macaw-aarch32/src/Data/Macaw/ARM/Eval.hs

```diff
@@ -112,14 +115,41 @@ mkInitialAbsState _mem startAddr =
 lowBitSet :: MC.ArchSegmentOff ARM.AArch32 -> Bool
 lowBitSet addr = MC.clearSegmentOffLeastBit addr /= addr
 
-absEvalArchFn :: MA.AbsProcessorState (MC.ArchReg ARM.AArch32) ids
+-- | Perform the given division operation if both the dividend and divisor are
+-- singleton abstract values (i.e., concrete)
+--
+-- NOTE: This could be extended to handle the case where both are 'MA.FinSet'
+-- values if we wanted to be fancier, but it doesn't seem necessary yet.
+--
+-- NOTE: The divisor is guaranteed by the ARM semantics to not be zero, but we
+-- guard against it here just in case, as the bv-sized operations *can* throw.
+abstractDivision
+  :: (BVS.BV w -> BVS.BV w -> BVS.BV w)
+  -> (BVS.BV w -> Integer)
+  -> NatRepr w
+  -> MA.AbsValue 32 (MT.BVType w)
+  -> MA.AbsValue 32 (MT.BVType w)
+  -> MA.AbsValue 32 (MT.BVType w)
+abstractDivision op extract wrep dividends divisors
```

Comment:
Well, they are abstract values that can represent multiple possible values.

Reply:
I see. In that case, it's perfectly accurate for `absEvalArchFn` to use singular `dividend`, since it only becomes plural after applying `transferValue`.